### DISCUSSION TITLE
Semaphore Docker: Ansible update planned?

### DISCUSSION THREAD

**Discussion by cf-sewe** [semaphoreui/semaphore#1109] [Q&A]
As Ansible is currently on the pretty old 2.10 branch in the provided docker image, do you consider updating it to 2.13 or newer?

There are already issues in our environment, where the main Ansible controllers are on 2.13 and the Playbooks are written and tested for that version. For example the notify on blocks feature was introduced only in ansible-core 2.11.

And another small thing which I just noticed, how are Python module requirements usually handled with Semaphore?
Like custom pip module installs (for example `dnspython` for ansible dig lookup)?

**Comment by dsmak** [comment]
Hi @cf-sewe, are you using snap-installed Semaphore? I'm looking to install pymysql for mysql module, and was wondering the same re pip installs.

**Reply by cf-sewe** [reply]
No, I am using the official Docker image provided by the Semaphore team.

As I am primarily running maintenance scripts via Semaphore, I have no need to install custom python modules within the Semaphore docker container. However I believe it should be possible it pip3 was installed in the semaphore container - which is isn't.

**Comment by andygeorge** [comment]
Edit: this could become a non-issue if [this PR](https://github.com/ansible-semaphore/semaphore/pull/1250) is merged

I had the same problem and ended up making a barebones container that uses `pip` to install Ansible, which allows for basically all versions. This is what I'm running at home, and it has been a drop-in replacement thus far.

Container: `ghcr.io/andygeorge/semaphore-pip-ansible` (https://github.com/andygeorge/containers/pkgs/container/semaphore-pip-ansible)
Containerfile source: https://github.com/andygeorge/containers/tree/main/semaphore

Example `Docker-compose.yml`:
```containerfile
services:
  semaphore:
    name: semaphore
    ports:
      - 3000:3000
    image: ghcr.io/andygeorge/semaphore-pip-ansible:latest # alias for semaphore-pip-ansible:ansible-8.3.0
    # image: docker.io/semaphoreui/semaphore:latest
```

On a running container:

```
$ podman exec -it semaphore sh
~ $ semaphore version
v2.9.4

~ $ ansible --version
ansible [core 2.15.3]
  config file = None
  configured module search path = ['/home/semaphore/.local/lib/python3.10/site-packages/ansible/modules']
  ansible python module location = /home/semaphore/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/semaphore/.local/lib/python3.10/site-packages/ansible/collections
  executable location = /home/semaphore/.local/bin/ansible
  python version = 3.10.13 (main, Aug 26 2023, 11:41:25) [GCC 11.2.1 20220219] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True

~ $ pip list|grep ansible
ansible            8.3.0
ansible-core       2.15.3
```

**Comment by tboerger** [comment]
We really need a proper process for installing Python and also Galaxy dependencies. The approach by @andygeorge is already a start, but it needs to be extended.
